src/StreamJsonRpc/JsonRpcEventSource.cs

```diff
@@ -0,0 +1,232 @@
+        public void SendNotificationStart(string message)
+        {
+            this.WriteEvent(SendNotificationStartEvent, message);
+        }
+
+        /// <summary>
+        /// Send Notification end.
+        /// </summary>
+        /// <param name="message">Notification details.</param>
+        [Event(SendNotificationStopEvent, Level = EventLevel.Verbose)]
+        public void SendNotificationStop(string message)
+        {
+            this.WriteEvent(SendNotificationStopEvent, message);
+        }
+
+        /// <summary>
+        /// Send Request start.
+        /// </summary>
+        /// <param name="message">Request details.</param>
+        [Event(SendRequestStartEvent, Level = EventLevel.Verbose)]
+        public void SendRequestStart(string message)
+        {
+            this.WriteEvent(SendRequestStartEvent, message);
+        }
+
+        /// <summary>
+        /// Send Request end.
+        /// </summary>
+        /// <param name="message">Request details.</param>
+        [Event(SendRequestStopEvent, Level = EventLevel.Verbose)]
+        public void SendRequestStop(string message)
+        {
+            this.WriteEvent(SendRequestStopEvent, message);
+        }
+
+        /// <summary>
+        /// Invoke Method start.
+        /// </summary>
+        /// <param name="message">Request and method details.</param>
+        [Event(InvokeMethodStartEvent, Level = EventLevel.Verbose)]
+        public void InvokeMethodStart(string message)
+        {
+            this.WriteEvent(InvokeMethodStartEvent, message);
+        }
+
+        /// <summary>
+        /// Invoke Method end.
+        /// </summary>
+        /// <param name="message">Request and method details.</param>
+        [Event(InvokeMethodStopEvent, Level = EventLevel.Verbose)]
+        public void InvokeMethodStop(string message)
+        {
+            this.WriteEvent(InvokeMethodStopEvent, message);
+        }
+
+        /// <summary>
+        /// Cancel Request start.
+        /// </summary>
+        /// <param name="message">Cancellation message.</param>
+        [Event(CancelRequestStartEvent, Level = EventLevel.Verbose)]
+        public void CancelRequestStart(string message)
+        {
+            this.WriteEvent(CancelRequestStartEvent, message);
+        }
+
+        /// <summary>
+        /// Cancel Request end.
+        /// </summary>
+        /// <param name="message">Cancellation message.</param>
+        [Event(CancelRequestStopEvent, Level = EventLevel.Verbose)]
+        public void CancelRequestStop(string message)
+        {
+            this.WriteEvent(CancelRequestStopEvent, message);
+        }
+
+        /// <summary>
+        /// Invoke Notification.
+        /// </summary>
+        /// <param name="message">Details of the Notification.</param>
+        [Event(InvokeNotificationStartEvent, Level = EventLevel.Verbose)]
+        public void InvokeNotificationStart(string message)
```

Comment:
start and stop events should be properly attributed as such so that PerfView recognizes the pairing. See https://github.com/microsoft/vs-threading/blob/90fc3eaff4add936f1a990de26c541a681605253/src/Microsoft.VisualStudio.Threading/ThreadingEventSource.cs#L86 for an example.

Reply:
Thanks. Did that.